chef/cookbooks/provisioner/recipes/base.rb

```diff
@@ -359,10 +366,18 @@
   source "crowbarrc.erb"
   variables(
     server: server,
+    username: username,
     password: password,
     verify_ssl: verify_ssl
   )
   owner "root"
   group "root"
   mode "0o600"
 end
+
+# Remove installer key, otherwise all root users could use it to talk to the crowbar server
+unless is_admin || ["hardware-installing", "installed", "readying"].include?(node[:state])
```

Comment:
I haven't checked, but why do we still need the file in these states? Is it because of some scripts that run (during autoyast/crowbar_register)? Would it be possible to do a followup (no need to block things here) to remove the need for this?

Reply:
crowbar_join needs the key in place during these states. I rather doubt it would be possible to undo that.